src/clp_ffi_js/ir/StreamReaderDataContext.hpp

```diff
@@ -5,28 +5,26 @@
 #include <utility>
 
 #include <clp/Array.hpp>
-#include <clp/ir/LogEventDeserializer.hpp>
-#include <clp/ir/types.hpp>
-#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/ReaderInterface.hpp>
 
 namespace clp_ffi_js::ir {
 /**
  * The data context for a `StreamReader`. It encapsulates a chain of the following resources:
- * A `clp::ir::LogEventDeserializer` that reads from a
- * `clp::streaming_compression::zstd::Decompressor`, which in turn reads from a `clp::Array`.
- * @tparam encoded_variable_t Type of encoded variables encoded in the stream.
+ * A CLP deserializer class that reads from a `clp::ReaderInterface`, which in turn reads from a
+ * `clp::Array`.
+ * @tparam deserializer_t Type of deserializer.
```

Comment:
This should be `Deserializer` since it's a non-primitive. We only use `snake_case_t` for primitive type defs.